src/nupic/algorithms/AnomalyLikelihood.cpp

```diff
@@ -0,0 +1,592 @@
+  // to return and the last windowSize values to store for later.
+  UInt toCrop = min((unsigned long)this->averagedAnomaly.getMaxSize(), runningLikelihoods.size());
+  this->runningLikelihoods.insert(runningLikelihoods.end() - toCrop, likelihoods.begin(),likelihoods.end()); //append & crop
+  assert(this->runningLikelihoods.size() <= this->averagedAnomaly.getMaxSize());
+
+  auto filteredLikelihoods = filterLikelihoods_(likelihoods);
+
+  if (verbosity > 3) {
+    cout << "Number of likelihoods:"<< likelihoods.size() << endl;
+//    print("First 20 likelihoods:", likelihoods[0:min(20, len(likelihoods))])
+    cout << "Leaving updateAnomalyLikelihoods."<< endl;
+  }
+
+  return filteredLikelihoods;
+}
+
+
+vector<Real> estimateAnomalyLikelihoods(vector<Real> anomalyScores, UInt skipRecords=0, UInt verbosity=0) { //FIXME averagingWindow not used, I guess it's not a sliding window, but aggregating window (discrete steps)!
+  /**
+  Given a series of anomaly scores, compute the likelihood for each score. This
+  function should be called once on a bunch of historical anomaly scores for an
+  initial estimate of the distribution. It should be called again every so often
+  (say every 50 records) to update the estimate. 
+
+  :param anomalyScores: a list of records. Each record is a list with the
+                        following three elements: [timestamp, value, score]
+
+                        Example::
+
+                            [datetime.datetime(2013, 8, 10, 23, 0), 6.0, 1.0]
+
+                        For best results, the list should be between 1000
+                        and 10,000 records
+  :param averagingWindow: integer number of records to average over
+  :param skipRecords: integer specifying number of records to skip when
+                      estimating distributions. If skip records are >=
+                      len(anomalyScores), a very broad distribution is returned
+                      that makes everything pretty likely.
+  :param verbosity: integer controlling extent of printouts for debugging
+
+                      0 = none
+                      1 = occasional information
+                      2 = print every record
+
+  :returns: 3-tuple consisting of:
+
+            - likelihoods
+
+              numpy array of likelihoods, one for each aggregated point
+
+            - avgRecordList
+
+              list of averaged input records
+
+            - params
+              a small JSON dict that contains the state of the estimator
+
+  **/
+  if (verbosity > 1) {
+    cout << "In estimateAnomalyLikelihoods."<<endl;
+    cout << "Number of anomaly scores:" <<  anomalyScores.size() << endl;
+    cout << "Skip records="<<  skipRecords << endl;
+//    print("First 20:", anomalyScores[0:min(20, len(anomalyScores))])
+  }
+
+  assert(anomalyScores.size() >= 0); // "Must have at least one anomalyScore"
+  auto dataValues = anomalyScores; //FIXME the "data" should be anomaly scores, or raw values? 
```

Comment:
what should the `dataValues` be? raw values, raw anomaly scores, aggreggated anom scores (current), likelihoods,...?